libraries/chain/webassembly/runtimes/eos-vm-oc/executor.cpp

```diff
@@ -251,6 +251,7 @@ void executor::execute(const code_descriptor& code, memory& mem, host_context& c
 
    context.trx_context.checktime(); //catch any expiration that might have occurred before setting up callback
 
+   int64_t retval = 0;
```

Comment:
It is not clear to me what `retval` should be for `EOSVMOC_EXIT_CHECKTIME_FAIL`.

Reply:
Thanks for the observation.

The `retval` here is used to populate the return code of the `call()` host function, where the caller of the sync call uses it to decide whether to abort the call or to let it as a no-op. That is why the value of `retval` must come from the `sync_call()` entry point; in all other failures it must be set to default `0` and the caller would handle them in other means.

I have added a comment to the code here.

Reply:
actually this is very similar context to my other comment,
https://github.com/AntelopeIO/spring/pull/1525#discussion_r2094599332
if the VM exits due to checktime (`EOSVMOC_EXIT_CHECKTIME_FAIL` here) it seems like `context.trx_context.checktime();` _must_ throw, otherwise we have problems.

Reply:
https://github.com/AntelopeIO/spring/pull/1525/commits/e356fe0028d0fcd907d8294df6c09b861e6d5285